applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -157,7 +164,54 @@ void TwoFluidNavierStokes<TElementData>::CalculateLocalSystem(
                     ComputeGaussPointEnrichmentContributions(data, Vtot, Htot, Kee_tot, rhs_ee_tot);
                 }
 
-                CondenseEnrichment(data, rLeftHandSideMatrix, rRightHandSideVector, Htot, Vtot, Kee_tot, rhs_ee_tot);
+                if (rCurrentProcessInfo[SURFACE_TENSION]){
+                    // Surface tension coefficient is set in material properties
+                    const double surface_tension_coefficient = this->GetProperties().GetValue(SURFACE_TENSION_COEFFICIENT);
+                    Matrix int_shape_function, int_shape_function_enr_neg, int_shape_function_enr_pos;
+                    GeometryType::ShapeFunctionsGradientsType int_shape_derivatives;
+                    Vector int_gauss_pts_weights;
+                    std::vector<Vector> int_normals_neg;
+                    Vector gauss_pts_curvature;
+
+                    ComputeSplitInterface(
+                        data,
+                        int_shape_function,
+                        int_shape_function_enr_pos,
+                        int_shape_function_enr_neg,
+                        int_shape_derivatives,
+                        int_gauss_pts_weights,
+                        int_normals_neg,
+                        p_modified_sh_func);
+
+                    CalculateCurvature(
+                        int_shape_function,
+                        gauss_pts_curvature);
+
+                    SurfaceTension(
+                        surface_tension_coefficient,
+                        gauss_pts_curvature,
+                        int_gauss_pts_weights,
+                        int_shape_function,
+                        int_normals_neg,
+                        rRightHandSideVector);
+
+                    PressureGradientStabilization(
+                        data,
+                        int_gauss_pts_weights,
+                        int_shape_function_enr_pos,
+                        int_shape_function_enr_neg,
+                        int_shape_derivatives,
+                        Kee_tot,
+                        rhs_ee_tot);
+
+                    CondenseEnrichment(rLeftHandSideMatrix, rRightHandSideVector, Htot, Vtot, Kee_tot, rhs_ee_tot);
+
+                } else{
+                    // Without pressure gradient stabilization, volume ratio is checked during condensation
+                    // Also, without surface tension, zero pressure difference is penalized
+                    CondenseEnrichment(data, rLeftHandSideMatrix, rRightHandSideVector, Htot, Vtot, Kee_tot, rhs_ee_tot);
```

Comment:
I really don't like this. You are overloading the function but implementing a way different behavior in each definition. In order to avoid the if statements, I think it would be better to have a `CondenseEnrichmentSurfaceTension` or something of the sort.

Reply:
I changed the name to: MerelyCondenseEnrichment
However, I have no idea how to omit the if statement.